smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/providers/AbstractMediator.java

```diff
@@ -233,16 +240,35 @@ public Multi<? extends Message<?>> decorate(Multi<? extends Message<?>> input) {
         }
 
         for (PublisherDecorator decorator : getSortedInstances(decorators)) {
-            input = decorator.decorate(input, getConfiguration().getOutgoing(), false);
+            input = decorator.decorate(input, getConfiguration().getOutgoings(), false);
         }
 
-        if (configuration.getBroadcast()) {
-            return BroadcastHelper.broadcastPublisher(input, configuration.getNumberOfSubscriberBeforeConnecting());
+        if (getBroadcast()) {
+            if (configuration.hasTargetedOutput() && !configuration.production().isMessageType()) {
+                input = input.map(s -> {
+                    TargetedMessages messages = TargetedMessages.from((Targeted) s.getPayload());
+                    return Messages.chain(s).with(messages.getPayload());
+                });
+            }
+            return BroadcastHelper.broadcastPublisher(input, getNumberOfSubscriberBeforeConnecting());
         } else {
             return input;
         }
     }
 
+    boolean getBroadcast() {
+        return configuration.getBroadcast() || configuration.getOutgoings().size() > 1;
+    }
+
+    int getNumberOfSubscriberBeforeConnecting() {
```

Comment:
That's something I want for @Broadcast too. I think I will query the graph to get the number of outgoing and handle ack consistently.